### PR TITLE
Adapted the HTTPlug integration docs to Async Client

### LINKED ISSUE #12472: [HttpClient] try using php-http/discovery when nyholm/psr7 is not insta…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#33980
| PR author(s) | @nicolas-grekas
| Merged in    | 4.4

### LINKED ISSUE #12475: [HttpClient] add HttpClient::createForBaseUri()
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#33973
| PR author(s) | @nicolas-grekas
| Merged in    | 4.4

### PR DESCRIPTION
**Pull request by Nyholm**
This should be merged with https://github.com/symfony/symfony/pull/33743

Closes #12472
Closes #12475

### CODE REVIEW

File: components/http_client.rst [hunk 1]
```diff
@@ -745,20 +745,15 @@ HTTPlug
...
-the ``HttplugClient`` adapter class. Similarly to ``Psr18Client`` implementing
-relevant parts of PSR-17, ``HttplugClient`` also implements the factory methods
-defined in the related ``php-http/message-factory`` package.
-
-Internally, the implementation relies on the ``Psr18Client``, so that the
-``psr/http-client`` package is needed to use this class:
+the :class:`Symfony\\Component\\HttpClient\\HttplugClient` adapter class. Similarly
+to ``Psr18Client`` implementing relevant parts of PSR-17, ``HttplugClient`` also
+implements the factory methods defined in the related ``php-http/message-factory``
+package.
```
Comment (Nyholm) [inline components/http_client.rst]: I dint not change the lines above but adding the class reference.

File: components/http_client.rst [hunk 2]
```diff
@@ -745,20 +745,15 @@ HTTPlug
 The `HTTPlug`_ specification was published before PSR-18 and is superseded by
 it. As such, you should not use it in newly written code. Yet, many libraries
 still require v1 or v2 of it. The component is interoperable with them thanks to
-the ``HttplugClient`` adapter class. Similarly to ``Psr18Client`` implementing
-relevant parts of PSR-17, ``HttplugClient`` also implements the factory methods
-defined in the related ``php-http/message-factory`` package.
```
Comment (Nyholm) [inline components/http_client.rst]: I did not change these 3 lines. I just added a class reference to `HttplugClient`

**Comment by nicolas-grekas** [comment]
Should we add some words about the async method? This one is not superseded by PSR18 and provides a promise-based API, which is a paradigm that could be worth mentioning?

**Review by nicolas-grekas — APPROVED** [review]
I pushed here the doc for all new features of HttpClient in 4.4

**Comment by Nyholm** [comment]
Thank you for completing this PR. 

I’m :+1:

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you both :pray:

**Pull request merged**